### PR TITLE
ci: init PR checks

### PR DESCRIPTION
**Pull request by pskfyi**

### CODE REVIEW

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: PR Checks
+
+on: pull_request
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:3]: Originally I set this to `on: [push, pull_request]` following a guide I found on Google, but when examining [the Actions tab](https://github.com/pskfyi/handy/actions) I found that those settings were causing the checks to run multiple times! Oof. This setting instead should cause checks to run only when a PR is created, updated, or reopened after being closed.

File: .github/workflows/pr-checks.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  pr-checks:
+    runs-on: macos-latest
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:7]: We'll expand this later to include Linux and Windows. Currently the test suite has only been fully checked against MacOS locally.

File: .github/workflows/pr-checks.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    steps:
+      - name: Clone the Repository
+        uses: actions/checkout@v3
+
+      - name: Install Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.x
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:16]: Using latest Deno. We might consider pinning a specific version number in the future.

File: .github/workflows/pr-checks.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+on: pull_request
+
+jobs:
+  pr-checks:
+    runs-on: macos-latest
+
+    steps:
+      - name: Clone the Repository
+        uses: actions/checkout@v3
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:11]: https://github.com/actions/checkout/tree/v3/

File: .github/workflows/pr-checks.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+      - name: Install Deno
+        uses: denoland/setup-deno@v1
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:14]: https://github.com/denoland/setup-deno

File: .github/workflows/pr-checks.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        with:
+          deno-version: v1.x
+
+      - name: Lint
+        run: deno lint
```
Comment (pskfyi) [inline .github/workflows/pr-checks.yml:19]: Only linting in this PR. Future checks will follow in subsequent PRs. For this PR, we should just ensure that the structure of the file is appropriate and mutually understood.

**Comment by pskfyi** [comment]
I have a Udemy course on GitHub Actions which I last watched in 2020. I'm going to skim through it again before marking this PR as ready for review.

**Comment by pskfyi** [comment]
Removed that commit to get checks to pass again.

**Review by AustinArey — APPROVED** [review]
Looks great, thanks for explaining everything in detail.

**Pull request merged**